integration/service/create_test.go

```diff
@@ -309,6 +312,144 @@ func TestCreateServiceConfigFileMode(t *testing.T) {
+		assert.Check(t, is.Equal(len(tasks), 1))
+
+		// just because the container started, doesn't mean it's actually run
+		// the command yet. if we try to read the logs instantly, we might be
+		// too fast and read before the output of the cat is available. to
+		// avoid that case, we're using poll.WaitOn with a closure to wait
+		// until logs actually respond
+		poll.WaitOn(t, func(t poll.LogT) poll.Result {
+			body, err := client.ContainerLogs(
+				ctx, tasks[0].Status.ContainerStatus.ContainerID,
+				types.ContainerLogsOptions{ShowStdout: true},
+			)
+			if err != nil {
+				return poll.Error(err)
+			}
+			defer body.Close()
+			// when we get logs, they come multiplexed into a single stream, for
+			// both stdout and stderr. in order to properly check the return value,
+			// we need to demultiplex these streams using stdcopy. The first arg is
+			// a bytes buffer we're creating to get everything out. The second arg
+			// is for stderr, which we don't care about and are throwing away. The
+			// third arg is the response we just got that we're reading from.
+			buf := &bytes.Buffer{}
+			if _, err := stdcopy.StdCopy(buf, ioutil.Discard, body); err != nil {
+				return poll.Error(err)
+			}
+
+			// get the string representation of the buffer, and trim off any extra
+			// whitespace that may have gotten in.
+			val := strings.TrimSpace(buf.String())
+
+			// if the value is empty, we may not have gotten the output of the
+			// container logs yet, and we should try again.
+			if val == "" {
+				return poll.Continue("output of container logs is empty")
+			}
+
+			// validate that the container logs, when the leading and trailing
+			// whitespace is removed, is equal to the expected expected.
+			// expected.
+			if val != expected {
+				// if they are not equal, then the test has failed and will
+				// never converge.
+				//
+				// the error returned includes a conversion to bytes because
+				// when i was writing this test, I kept getting messages like
+				// "0" != "0" and didn't realize until after I saw the byte
+				// output what had happened (it was the aforementioned stdcopy
+				// logic not being present)
+				return poll.Error(fmt.Errorf(
+					"%q != %q: actual value is: %v, expected value is: %v",
+					val, expected, []byte(val), []byte(expected),
+				))
+			}
+
+			// otherwise, if the value and the expected value match, we're
+			// done.
+			return poll.Success()
+		})
+
+		// verify that the task has the sysctl option set in the task object
+		assert.DeepEqual(t, tasks[0].Spec.ContainerSpec.Sysctls, sysctlOpts)
+
+		// inspect the service to verify that it has the sysctl option in the
+		// spec.
+		service, _, err := client.ServiceInspectWithRaw(ctx, serviceID, types.ServiceInspectOptions{})
+		assert.NilError(t, err)
+		assert.DeepEqual(t,
+			service.Spec.TaskTemplate.ContainerSpec.Sysctls, sysctlOpts,
+		)
+
+		// remove the service
+		err = client.ServiceRemove(ctx, serviceID)
```

Comment:
Services should already be removed when the test completes, so I think we can skip this, and just continue with the next iteration